wallet/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -784,6 +823,17 @@ flushWalletStorage = modify flushDo
                             , _wiIsReady   = False
                             }
 
+-- | Unsafe address conversion for use in migration. This will throw an error if
+--   the address cannot be migrated.
+unsafeCIdToAddress :: WebTypes.CId WebTypes.Addr -> Address
+unsafeCIdToAddress cId = case WebTypes.cIdToAddress cId of
+    Left err -> error $ "unsafeCIdToAddress: " <> err
+    Right x  -> x
+
+cwamToWam :: WebTypes.CWAddressMeta -> WAddressMeta
+cwamToWam (WebTypes.CWAddressMeta wid accIdx addrIdx cAddr) =
+    WAddressMeta wid accIdx addrIdx $ unsafeCIdToAddress cAddr
```

Comment:
This can become the `migrate` method for the `Migrate` instance for `WAddressMeta`